provider/openstack/firewaller.go

```diff
@@ -49,6 +54,10 @@ type Firewaller interface {
 	// DeleteGroups deletes the security groups with the specified names.
 	DeleteGroups(names ...string) error
 
+	// Implementations are expected to update all of the security
```

Comment:
I realise the function below doesn't follow the pattern, but please stick to the idiom of:

"UpdateGroupController updates all of the security groups for this model to refer to the specified controller."

I think the intent might be a bit clearer by appending something like ", such that DeleteAllControllerGroups will remove them only when called with the specified controller ID."